core/json/src/main/scala/net/liftweb/json/Extraction.scala

```diff
@@ -333,22 +333,25 @@ object Extraction {
 
     def newTuple(root: JValue, mappings: List[Mapping]): Any = {
       root match {
-        case JArray(items) if items.length >= 1 && items.length <= 22 =>
+        case JArray(items) if items.nonEmpty && items.length <= tuples.length =>
           val builtItems: Seq[Object] = items.zip(mappings).map({
-            case (i,m) =>
-              build(i, m).asInstanceOf[Object]
+            case (item, mapping) =>
+              build(item, mapping).asInstanceOf[Object]
           })
-          val tupleClass = Class.forName("scala.Tuple" + items.length)
+          val tupleIndex = items.length - 1
+          val tupleClass = tuples.lift(tupleIndex).getOrElse {
+            throw new IllegalArgumentException(s"Cannot instantiate a tuple of length ${items.length} even though that should be a valid tuple length.")
```

Comment:
> I'm wondering if we just do a .get here and point to the initialization of tuples to show that if we've gotten to this point in the code, we know that entry exists.

No, I think having a `getOrElse` with an exception here is a good way of refactor-proofing the code. Having worked in a project that made way-too-ample use of `Option.get` last year I think I've landed on always justifying a retrieval that has no alternative much like `Box`'s `openOrThrowException` requires us to do. The alternative is making a bet that this code can't be fouled by an accidental fat finger and then surfacing an opaque error message to a user of the Framework. Certainly, surfacing this error at all would be bad, but making it opaque is just insult to injury. :)

> Still think it may be worth turning tuples into a Map (only doing the .getConstructors()(0) bit once during the singleton intialization).

I'm not quite following the entirety of this.